docs/user_guide/fairness_in_machine_learning.rst

```diff
@@ -141,6 +141,132 @@ which groups these values were observed, as well as the difference and ratio
+
+The Solutionism Trap
+^^^^^^^^^^^^^^^^^^^^
+
+This trap occurs when we assume that the best solution to a problem
+involves technology, and fail to recognize other possible solutions outside of
+this realm. One area where this manifests in is within contexts in which the
+definition "fairness" changes or is dependent on a political context.
+
+Example: consider the problem of internet connectivity in rural communities.
+An example of the solutionism trap is assuming that by using data science to
+study internet speed in a given region, policies and deals can be arranged to
+reduce the price of internet access. However, if there are additional socioeconomic
+problems within a community, for example lack of education, infrastructure, information
+technology and health services, then an algorithmic solution purely focused on internet
+speed will fail to meaningfully address the needs of the community.
+
+
+The Ripple Effect Trap
+^^^^^^^^^^^^^^^^^^^^^^
+
+Selbst et al. [#4]_ define this as a "failure to understand how the insertion of
+technology into an existing social system changes the behaviors and embedded values
+of the pre-existing system".
+
+Example 1:
+
+The Formalism Trap
+^^^^^^^^^^^^^^^^^^
+
+This trap occurs when implementing an algorithmic solution that fails to take
+into account the social dimensions associated to fairness in a situation. These
+dimensions include "procedurality", "contextuality" and "contestability". Such
+dimensions can often not be resolved via a purely mathematical framework.
+
+Example:
+
+The Portability Trap
+^^^^^^^^^^^^^^^^^^^^
+
+This trap occurs when we fail to understand how reusing a model or
+algorithm that is designed for one specific social context, may not necessarily
+apply to a different social context. Reusing an algorithmic solution and failing
+to take into account differences in involved social contexts can result in misleading
+results and potentially harmful consequences if the algorithm is used to determine the
+fate of an individual.
+
+Example: Reusing a machine learning algorithm used to screen job applications in the
+nursing industry, for job applications in the information technology sector. An intuitive
+yet important difference between both contexts is the difference in skills required to
+succeed in both industries. A slightly more subtle difference is the demographic differences
+in gender attraction to each of the industries, resulting from wording in job postings,
+social constructs on gender and societal roles, and the male-female ratio of successful
+applicants in each field.
+
+The Framing Trap
+^^^^^^^^^^^^^^^^
+
+This trap occurs when data scientists fail to consider the full picture surrounding
+a particular social context when abstracting a social problem, and implementing an
+algorithm in which the outcome involves enforcing decisions that will impact a person
+or group of people.
+
+The authors define three main frames:
```

Comment:
I love that you included these definitions in here! I wonder whether we can introduce it a bit for readers, something like:

"To help us avoid drawing narrow boundaries of what is considered in scope for the problem, we might consider different "frames" to use. The first, narrowest frame, focuses only on modeling choices made when designing the algorithm..."

Reply:
Incorporated sentence